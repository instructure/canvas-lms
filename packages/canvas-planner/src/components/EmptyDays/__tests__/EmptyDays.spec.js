/*
 * Copyright (C) 2018 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react'
import {shallow} from 'enzyme'
import MockDate from 'mockdate'
import {EmptyDays} from '../index'

const TZ = 'Asia/Tokyo'

const getDefaultProps = (overrides = {}) => {
  return {
    day: '2017-04-23',
    endday: '2017-04-26',
    animatableIndex: 0,
    timeZone: TZ,
    registerAnimatable: () => {},
    deregisterAnimatable: () => {},
    ...overrides
  }
}

beforeAll(() => {
  MockDate.set('2017-04-22', TZ)
})

afterAll(() => {
  MockDate.reset()
})

it('renders the component', () => {
  const wrapper = shallow(<EmptyDays {...getDefaultProps()} />)
  expect(wrapper).toMatchSnapshot()
})

it('renders the today className when surrounds Today', () => {
  const wrapper = shallow(<EmptyDays {...getDefaultProps({day: '2017-04-22'})} />)
  expect(wrapper).toMatchSnapshot()
})
