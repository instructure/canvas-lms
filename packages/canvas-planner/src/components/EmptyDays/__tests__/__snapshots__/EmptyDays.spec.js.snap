// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  className="EmptyDays-styles__root planner-empty-days"
>
  <Heading
    border="bottom"
    color="inherit"
    level="h2"
  >
    <Text
      as="div"
      letterSpacing="normal"
      lineHeight="condensed"
      size="medium"
      wrap="normal"
    >
      April 23 to April 26
    </Text>
  </Heading>
  <div
    className="EmptyDays-styles__nothingPlannedContent"
  >
    <GroupedDates
      aria-hidden="true"
      role="img"
      viewBox="0 0 907 155"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    />
    <div
      className="EmptyDays-styles__nothingPlannedContainer"
    >
      <div
        className="EmptyDays-styles__nothingPlanned"
      >
        <Text
          as="span"
          letterSpacing="normal"
          size="large"
          wrap="normal"
        >
          Nothing Planned Yet
        </Text>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the today className when surrounds Today 1`] = `
<div
  className="EmptyDays-styles__root planner-empty-days planner-today"
>
  <Heading
    border="bottom"
    color="inherit"
    level="h2"
  >
    <Text
      as="div"
      letterSpacing="normal"
      lineHeight="condensed"
      size="medium"
      wrap="normal"
    >
      April 22 to April 26
    </Text>
  </Heading>
  <div
    className="EmptyDays-styles__nothingPlannedContent"
  >
    <GroupedDates
      aria-hidden="true"
      role="img"
      viewBox="0 0 907 155"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    />
    <div
      className="EmptyDays-styles__nothingPlannedContainer"
    >
      <div
        className="EmptyDays-styles__nothingPlanned"
      >
        <Text
          as="span"
          letterSpacing="normal"
          size="large"
          wrap="normal"
        >
          Nothing Planned Yet
        </Text>
      </div>
    </div>
  </div>
</div>
`;
